components/brave_rewards/resources/extension/brave_rewards/background/reducers/rewards_panel_reducer.ts

```diff
@@ -25,6 +25,49 @@ const getWindowId = (id: number) => {
   return `id_${id}`
 }
 
+function pollTwitchPage (tab: chrome.tabs.Tab, tabId: number, publisherBlob:string): void {
+  let url = tab.url
+  // use an interval here to monitor when the DOM has finished
+  // generating. clear after the data is present.
+  // Check every second no more than 'limit' times
+  // clear the interval if panel closes
+  const markupMatch = '<figure class=\"tw-avatar tw-avatar--size-36\">' +
+                      '<div class=\"tw-border-radius-medium tw-overflow-hidden\">' +
+                      '<img class=\"tw-avatar__img tw-image\" alt=\"'
+  let itr = 0
+  const limit = 10
+
+  let interval = setInterval(function () {
+    chrome.tabs.executeScript(tabId, {
+      code: 'document.body.outerHTML'
+    }, function (result: string[]) {
+      if (result[0].includes(markupMatch)) {
+        publisherBlob = result[0]
+        clearInterval(interval)
+        if (url) {
+          chrome.braveRewards.onTwitchInfo(publisherBlob, url)
+        }
+        if (tab && tab.windowId && tab.url && tab.favIconUrl) {
+          chrome.braveRewards.getTwitchPublisherData(tab.windowId, tab.url, tab.favIconUrl || '', publisherBlob)
+        }
+      } else {
+        chrome.storage.local.get(['rewards_panel_open'], function(result) {
+          if (result['rewards_panel_open'] === 'false') {
+            // panel was closed. give up
+            clearInterval(interval)
+          }
+        })
+        itr++
+        if (itr === limit) {
+          // give up
+          clearInterval(interval)
+        }
+      }
+    })
+    return arguments.callee
```

Comment:
haven't seen this pattern before, but it seems to work. the main downside is that according to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/arguments/callee, arguments.callee cannot be invoked in strict mode, and we recommend that js files use strict mode.

is it ok for the first polling to just happen 1s later?

Reply:
So would simply giving the function a name and return the name comply with strict?

Also, if the polling is delayed, the user sees Twitch.tv in the panel for at least one second rather than the publisher.

Reply:
example:
```
let interval = setInterval(function poll() {
...
...
   return poll
}(), 1000)```

Reply:
@jasonrsadler that should be fine. this would also work:
```
function poll() {...}
poll()
let interval = setInterval(poll, 1000)
```

Reply:
Done.